From fcdc0aecbad584c21f089e2b27432383250bf673 Mon Sep 17 00:00:00 2001
From: "Tian, Shilei" <Shilei.Tian@amd.com>
Date: Mon, 14 Apr 2025 15:02:17 -0400
Subject: [PATCH 7/8] Policy `CMP0053` is no longer supported by CMake 4+
 (#1519)

The part of the code needs to be refined; otherwise we will have
multiple breaks
after users upgrade to CMake 4 on Windows.
---
 amd/device-libs/cmake/OCL.cmake | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/amd/device-libs/cmake/OCL.cmake b/amd/device-libs/cmake/OCL.cmake
index 4f16d8cd81d7..1dabde51b8c0 100644
--- a/amd/device-libs/cmake/OCL.cmake
+++ b/amd/device-libs/cmake/OCL.cmake
@@ -9,7 +9,10 @@
 # command-lines, but the only way to do this as part of the dependency graph is
 # configure_file and we are included from multiple places. To get around this
 # we `file(WRITE)` a file with an @variable reference and `configure_file` it.
-cmake_policy(SET CMP0053 OLD)
+# FIXME: CMP0053 is removed in CMake 4; refine code relying on this policy.
+if(${CMAKE_VERSION} VERSION_LESS "4.0.0")
+  cmake_policy(SET CMP0053 OLD)
+endif()
 
 if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.20.0")
   # The policy change was for handling of relative paths for
-- 
2.48.1

